test/highlighter/test.js

```diff
@@ -1050,4 +1069,505 @@ define(['jquery', 'lodash', 'ui/highlighter'], function($, _, highlighterFactory
 
         assert.equal($(fixtureContainer).find('.hl').length, 0, 'Third higlight has been discarded');
     });
+
+    QUnit.module('highlighter with multi colors');
+
+    QUnit.cases
+        .init([
```

Comment:
To improve readability I would move this huge array up into as a const, like is done with `highlightRangeData`. Or even better move both to a different file and import them from there.

Reply:
I don’t think to decouple test cases from the test it's a good idea and will improve readability

Reply:
**thought:** Indeed, the data fixtures could have been moved to an external JSON resource, and the dynamic code  could have been generated based on properties. Anyway, this is not a big issue.